pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -147,7 +150,7 @@ func ensureTeamState(workflowCtx *workflow.Context, team *akov2.AtlasTeam) (stri
 			return "", workflow.Terminate(workflow.TeamNotUpdatedInAtlas, err.Error())
 		}
 
-		return atlasTeam.ID, workflow.OK()
+		return *atlasTeam.Id, workflow.OK()
```

Comment:
https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_5_best_practices.md#using-getters-instead-of-direct-field-access